src/services/recipient.js

```diff
@@ -805,39 +804,13 @@ export async function getGoalsByActivityRecipient(
     return goal;
   });
 
+  // reduce
   const r = sorted.reduce((previous, current) => {
```

Comment:
This should be a map now, not a reduce, since it is transforming an array of objects into an array of the same length/modified objects. Additionally, since there is a map called right before this reduce, you could combine the two iterations and gain a little performance.

Reply:
Thinking a little more about this, I think you could now do the sorting in SQL but I could be wrong. Probably stuck doing the limit/offset in JavaScript though.